frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/local-volume-set-inner.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import {
+  FormGroup,
+  TextInput,
+  Radio,
+  Expandable,
+  TextInputTypes,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { Dropdown } from '@console/internal/components/utils';
+import { ListPage } from '@console/internal/components/factory';
+import { k8sList, NodeKind } from '@console/internal/module/k8s';
+import { getName } from '@console/shared';
+import { NodeModel } from '@console/internal/models';
+import { hasTaints } from '@console/ceph-storage-plugin/src/utils/install';
+import { NodesSelectionList } from './nodes-selection-list';
+import {
+  State,
+  Action,
+  diskModeDropdownItems,
+  diskTypeDropdownItems,
+  diskSizeUnitOptions,
+} from './state';
+import './create-local-volume-set.scss';
+
+let allNodeNames: string[] = [];
```

Comment:
Its used at two places inside the component. Declaring inside will re-initialize it again and again on each render which is not required.

Reply:
State's persist between re-render use `useState`

Reply:
declaring this as state is not required as here as its will be updated just 1st time and later on it will be used and no updates required. Hence, declaring it outside of component makes sense here.

Reply:
When you update this variable react will not re-render since it's neither a state variable or not props. It is not a good practice to declare variables used by a component outside of its scope. There are better ways to achieve the functionality you desire than by declaring it as a global.